docs/UserGuide.md

```diff
@@ -141,80 +141,122 @@ Format: `clear`
 Adds a supplier to the address book.
 
 Format: `add -s n/NAME p/PHONE e/EMAIL com/COMPANY [t/TAG]…​ [pro/PRODUCT]…​`
-<box type="tip" seamless>
 
-Parameters:
+<box type="details" seamless>
 
-- `n/NAME`: Must be alphanumeric, and must not be blank.
-- `p/PHONE`: Must be numeric, must contain at least 3 digits, and must not be blank.
-- `e/EMAIL`: Must be a valid email address, and must not be blank.
-- `com/COMPANY`: Must be alphanumeric, and must not be blank.
-- `[t/TAG]`: Must be alphanumeric, and must be between 1 and 50 (inclusive) characters long.
-- `[pro/PRODUCT]`: Must be alphanumeric, can include spaces, and must be between 1 and 50 (inclusive) characters long.
+Parameters:
 
-**Tip:** A supplier can have any number of tags and products (including 0)
+- `n/NAME`: The supplier's name. It must be alphanumeric, and cannot be blank.
+- `p/PHONE`: The supplier's phone number. It must be numeric, and contain at least 3 digits.
+- `e/EMAIL`: The supplier's email address. It must be in a valid email address format, and cannot be blank.
+  - Please see [below](#valid-email-address-format) for more information on what constitutes a valid email address format.
+- `com/COMPANY`: The company associated with the supplier. It must be in a valid company name format, and cannot be blank.
+  - Please see [below](#valid-company-name-format) for more information on what constitutes a valid company name format.
+- `[t/TAG]`: The tag(s) associated with the supplier. It must be alphanumeric, and only contain between 1 and 50 (inclusive) characters.
+- `[pro/PRODUCT]`: The product(s) associated with the supplier. It must be alphanumeric, only contain between 1 and 50 (inclusive) characters, and spaces are also allowed.
+  - One space is counted as one character.
 
-**Tip:** A supplier has a default status of `active`
+**Tip:** A supplier can have any number of tags and products (including 0).
 
 </box>
 
 <box type="warning" seamless>
 
 **Warnings**:
-- At least one space between `add` and `-s`
-- Duplicate supplier will not be added again
-- A supplier is considered duplicate if they have the same `NAME` and `COMPANY`
-- Adding duplicate `TAG`/`PRODUCT` will result in only one copy added to the supplier
-- A tag/product is considered duplicate if they have the same `TAG`/`PRODUCT`
+- At least one space is needed between `add` and `-s`.
+- A warning will be given if the user tries to add a duplicate supplier.
+- A supplier is considered duplicate if they have the same `NAME` and `COMPANY`.
+  - Comparison between different `NAME`/`COMPANY` is case-sensitive.
```

Comment:
Our current implementation is such that name is case sensitive but company is not case sensitive.